spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1357,280 @@ Union type extensions have the potential to be invalid if incorrectly defined.
+  to the input coercion rules for the type of the tagged member for that field.
+
+* If a variable is provided for an input object field, the runtime value of that
+  variable must be used. If the runtime value is {null} and the tagged member
+  type is non-null, a field error must be thrown. If no runtime value is
+  provided, the variable definition's default value should be used. If the
+  variable definition does not provide a default value, the field should be
+  treated as if it was not specified.
+
+* No entry should be added to the unordered map for any other members defined
+  on the Tagged type.
+
+Following are examples of input coercion for a Tagged type with a
+`String` member `a` and a required (non-null) `Int!` member `b`:
+
+```graphql example
+tagged ExampleInputTagged {
+  a: String
+  b: Int!
+}
+```
+
+Literal Value            | Variables               | Coerced Value
+------------------------ | ----------------------- | ---------------------------
+`{ a: "abc", b: 123 }`   | `{}`                    | Error: Exactly one key must be specified
+`{ a: null, b: 123 }`    | `{}`                    | Error: Exactly one key must be specified
+`{ b: 123 }`             | `{}`                    | `{ b: 123 }`
+`{ a: $var, b: 123 }`    | `{ var: null }`         | Error: Exactly one key must be specified
+`{ a: $var, b: 123 }`    | `{}`                    | `{ b: 123 }`
+`{ b: $var }`            | `{ var: 123 }`          | `{ b: 123 }`
+`$var`                   | `{ var: { b: 123 } }`   | `{ b: 123 }`
+`"abc123"`               | `{}`                    | Error: Incorrect value
+`$var`                   | `{ var: "abc123" } }`   | Error: Incorrect value
+`{ a: "abc", b: "123" }` | `{}`                    | Error: Exactly one key must be specified
+`{ b: "123" }`           | `{}`                    | Error: Incorrect value for member {b}
+`{ a: "abc" }`           | `{}`                    | `{ a: "abc" }`
+`{ b: $var }`            | `{}`                    | Error: No keys were specified
+`$var`                   | `{ var: { a: "abc" } }` | `{ a: "abc" }`
+`{ a: "abc", b: null }`  | `{}`                    | Error: Exactly one key must be specified
+`{ b: $var }`            | `{ var: null }`         | Error: {b} must be non-null.
+`{ b: 123, c: "xyz" }`   | `{}`                    | Error: Unexpected member {c}
+
+
+
+**Type Validation**
+
+Tagged types have the potential to be invalid if incorrectly defined. This set
+of rules must be adhered to by every Tagged type in a GraphQL schema.
+
+1. A Tagged type must define one or more members.
+2. For each member of a Tagged type:
+   1. The member must have a unique name within that Tagged type;
+      no two members may share the same name.
+   2. The member must not have a name which begins with the
+      characters {"__"} (two underscores).
+3. {IsOutputType(fieldType)} and {IsInputType(type)} cannot both be {false}.
```

Comment:
What do `fieldType` and `type` refer to? This part may be expanded a bit for clarity, without reading the rest of the document it is not immediately clear what it means.

Reply:
They should both be `taggedType`; good spot. The "Objects" section references `objectType` without further qualification; so I think this is fine (now it's fixed).